types/three/src/core/BufferAttribute.d.ts

```diff
@@ -43,6 +43,8 @@ export class BufferAttribute {
     set needsUpdate(value: boolean);
 
     readonly isBufferAttribute: true;
+    readonly isInterleavedBufferAttribute: false;
+    readonly isGLBufferAttribute: false;
```

Comment:
I don't think these `false` are explicitly set.

Reply:
True that. Should I rather set these as `undefined` instead of `false`? If we omit these fields completely then we no longer can use them to discriminate between the different attribute types.

Reply:
Ok that's the motivation, I got it.
You want to use `position.isGLBufferAttribute` here right?

https://github.com/three-types/three-ts-types/blob/9ce3aa43d8b9af2d63217cd9dfa177d7ae4edcbe/types/three/test/controls/controls-pointercontrols.ts#L129

But having `false` here only for this is lying since it isn't actually `false`.

You can solve this by either of these 3 solutions:

1. use `!(position instanceof GLBufferAttribute)` instead for the condition
2. use `'setXYZ' in position` instead
3. type assert the `position`, `let position = floorGeometry.attributes.position as BufferAttribute;`

I prefer the (3), since it's obvious that the `position` is `BufferAttribute` here since it's made by `new THREE.PlaneGeometry()`.

Reply:
To me it seems that it would be most natural to use the publicly documented boolean fields (e.g. https://threejs.org/docs/#api/en/core/BufferAttribute.isBufferAttribute) for checking this. So my suggestion would be to still  change the `false` values to `undefined` and then the types would precisely match the returned values. If we omit the non-true fields from TS types then it's not possible to refer to the boolean field unless you already have the corresponding attribute type. And then there's no longer any point in checking a trivially true field :)

The three solutions you listed above definitely do work and nothing prevents the user from choosing any of them. And I agree that in many cases a type assertion can be the most straightforward option. I just think that it would be better to support the documented fields in a meaningful way in TS, as well. WDYT?